pkg/serverless/trace/trace.go

```diff
@@ -130,9 +148,35 @@ func filterSpanFromLambdaLibraryOrRuntime(span *pb.Span) bool {
 			return true
 		}
 	}
+
+	// Filter out DNS traces
+
+	if val, ok := span.Meta[dnsAddressMetaKey]; ok {
+		if strings.HasPrefix(val, LocalHostNamePrefix) {
+			log.Debugf("Detected DNS span with locahost address %s, removing it", val)
+			return true
+		}
+
+		if strings.HasPrefix(val, dnsHostNamePrefix) {
+			log.Debugf("Detected DNS span with non-routable address %s, removing it", val)
+			return true
+		}
+	}
+
+	// Filter out TCP traces
+
+	if val, ok := span.Meta[tcpAddressMetaKey]; ok {
+		if strings.HasPrefix(val, LocalHosNamePrefix) {
+			log.Debugf("Detected TCP span with tcp adress %s, removing it", val)
+			return true
+		}
+	}
```

Comment:
At the moment, these conditions will filter out _all_ TCP + DNS spans with `127.0.0.1` or `0.0.0.0`. Let's narrow this logic down to include ports 8124 (DD extension), 8125 (dogstatsd) and 9001 (Lambda runtime).